app/features/accounts/account-selector.tsx

```diff
@@ -0,0 +1,109 @@
+import { ChevronDown, ChevronUp, Zap } from 'lucide-react';
+import { LandmarkIcon } from 'lucide-react';
+import { useState } from 'react';
+import {
+  Drawer,
+  DrawerContent,
+  DrawerHeader,
+  DrawerTrigger,
+} from '~/components/ui/drawer';
+import { formatUnit } from '~/lib/formatting';
+
+export type AccountType = 'cashu' | 'nwc';
+
+export type Account = {
+  id: string;
+  name: string;
+  unit: 'usd' | 'sat'; // in this case usd is cents
+  type: AccountType;
+  balance: number;
+} & (
+  | {
+      type: 'cashu';
+      mintUrl: string;
+    }
+  | {
+      type: 'nwc';
+      nwcUrl: string;
+    }
+);
+
+const CashuIcon = () => <LandmarkIcon className="h-4 w-4" />;
+const NWCIcon = () => <Zap className="h-4 w-4" />;
+
+const getAccountIcon = (type: AccountType) => {
```

Comment:
nit but there is more compact way to write stuff like this:
```
const icons: Record<AccountType, ReactNode> = {
  cashu: <CashuIcon />
  nwc: <NWCIcon />
}
```

then you can just do `icons[account.type]`